### ISSUE TITLE
[VersionControl] Override the commit author with empty email (Git)

### ISSUE THREAD

**Issue by jsuarezruiz** [mono/monodevelop#7103]
Committing changes, can override the user info with a custom name and email for the commit. There aren't any form validation. Leaving the email empty:

```
ERROR [2019-02-08 09:50:23Z]: Version control operation failed:
System.ArgumentNullException: Value cannot be null.
Parameter name: email
  at LibGit2Sharp.Core.Ensure.ArgumentNotNullOrEmptyString (System.String argumentValue, System.String argumentName) [0x00020] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/external/libgit2sharp/LibGit2Sharp/Core/Ensure.cs:56
  at LibGit2Sharp.Signature..ctor (System.String name, System.String email, System.DateTimeOffset when) [0x00014] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/external/libgit2sharp/LibGit2Sharp/Signature.cs:36
  at MonoDevelop.VersionControl.Git.GitRepository.OnCommit (MonoDevelop.VersionControl.ChangeSet changeSet, MonoDevelop.Core.ProgressMonitor monitor) [0x00098] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/addins/VersionControl/MonoDevelop.VersionControl.Git/MonoDevelop.VersionControl.Git/GitRepository.cs:863
  at MonoDevelop.VersionControl.Repository.Commit (MonoDevelop.VersionControl.ChangeSet changeSet, MonoDevelop.Core.ProgressMonitor monitor) [0x0005d] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl/Repository.cs:593
  at MonoDevelop.VersionControl.CommitCommand+CommitWorker.Run () [0x00081] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl/CommitCommand.cs:70
  at MonoDevelop.VersionControl.VersionControlTask.<BackgroundWorker>b__13_0 () [0x00000] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl/VersionControlTask.cs:51
  at System.Threading.Tasks.Task.InnerInvoke () [0x0000f] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-08/external/bockbuild/builds/mono-x64/external/corert/src/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs:2500
  at System.Threading.Tasks.Task.Execute () [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-08/external/bockbuild/builds/mono-x64/external/corert/src/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs:2343
```

> VS bug [#789300](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/789300)

**Comment by jsuarezruiz** [comment]
Sent PR: https://github.com/mono/monodevelop/pull/7130

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.0.0.1534 (master)

Pull Request #7130 merged by: Marius Ungureanu
Author: mono
Commit: 46372fc641951606c613ce3b956f6f6246b754e4 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.0.0.2306 (release-8.0)

Pull Request #7130 merged by: Marius Ungureanu
Author: mono
Commit: 46372fc641951606c613ce3b956f6f6246b754e4 (xamarin/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.6.0.802 (master)

Pull Request #7130 merged by: Marius Ungureanu
Author: mono
Commit: 46372fc641951606c613ce3b956f6f6246b754e4 (xamarin/monodevelop)
